### PR TITLE
Parse TextComponent hover values depending on action

### PR DESCRIPTION
**Pull request by srockw**
Also turn all string actions to lowercase, otherwise `byName` does not find them

### CODE REVIEW

File: src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt [hunk 1]
```diff
@@ -252,6 +262,26 @@ class TextComponent : Text {
         Message(this).actionBar()
     }
 
+    private fun parseItemContent(obj: Any?): HoverEvent.ItemStackContent? {
+        return when (obj) {
+            is ItemStack -> obj
+            is Item -> obj.toMC()
+            is String -> ItemType(obj).asItem().toMC()
+            is HoverEvent.ItemStackContent -> return obj
+            else -> null
```
Comment (mattco98) [inline src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt:271]: Let's throw an error in the else case instead of quietly doing nothing, same for parseEntityContent

File: src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt [hunk 2]
```diff
@@ -252,6 +262,26 @@ class TextComponent : Text {
         Message(this).actionBar()
     }
 
+    private fun parseItemContent(obj: Any?): HoverEvent.ItemStackContent? {
```
Comment (mattco98) [inline src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt:265]: This parameter will never be null, so lets remove the null marker here and in the `?.let`. Same for `parseEntityContent`.

File: src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt [hunk 3]
```diff
@@ -104,7 +109,7 @@ class TextComponent : Text {
     fun setClickAction(value: Any?) = apply {
         clickAction = when (value) {
             is ClickEvent.Action -> value
-            is String -> ClickEvent.Action.byName(value)
+            is String -> ClickEvent.Action.byName(value.lowercase())
```
Comment (mattco98) [inline src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt:112]: Can you put these in a separate commit? In general if you write "Also, do xyz" anywhere in your commit message, that's a good sign that whatever you're talking about should be in its own commit.

File: src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt [hunk 4]
```diff
@@ -190,9 +195,14 @@ class TextComponent : Text {
...
     fun setHoverValue(value: Any?) = apply {
-        hoverValue = if (hoverAction == HoverEvent.Action.SHOW_TEXT && value != null) {
-            from(value)
-        } else value
+        hoverValue = value?.let {
+            when (hoverAction) {
+                HoverEvent.Action.SHOW_TEXT -> from(value)
+                HoverEvent.Action.SHOW_ITEM -> parseItemContent(value)
+                HoverEvent.Action.SHOW_ENTITY -> parseEntityContent(value)
+                else -> value
```
Comment (mattco98) [inline src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt:203]: If the only other case is `null`, lets write that explicitly instead of using `else`. That way if another enum is added in a future version, it'll be a compiler error.

File: src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt [hunk 5]
```diff
@@ -190,9 +195,14 @@ class TextComponent : Text {
...
      */
     fun setHoverValue(value: Any?) = apply {
-        hoverValue = if (hoverAction == HoverEvent.Action.SHOW_TEXT && value != null) {
-            from(value)
-        } else value
+        hoverValue = value?.let {
+            when (hoverAction) {
+                HoverEvent.Action.SHOW_TEXT -> from(value)
+                HoverEvent.Action.SHOW_ITEM -> parseItemContent(value)
+                HoverEvent.Action.SHOW_ENTITY -> parseEntityContent(value)
```
Comment (mattco98) [inline src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt:202]:
```suggestion
                HoverEvent.Action.SHOW_ITEM -> parseItemContent(it)
                HoverEvent.Action.SHOW_ENTITY -> parseEntityContent(it)
```

**Review by mattco98 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt (hunk 4 above)
Comment (srockw) [reply src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt:203]: The only other case is not `null` due to `hoverAction` being of type `HoverEvent.Action<*>?`

Comment (mattco98) [reply src/main/kotlin/com/chattriggers/ctjs/api/message/TextComponent.kt:203]: Ahh is this the class that isn't an enum? In that case nvm
